src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -233,10 +236,14 @@ - (void)peripheral:(CBPeripheral *)peripheral
         chip::Ble::ChipBleUUID svcId;
         chip::Ble::ChipBleUUID charId;
         [BleConnection fillServiceWithCharacteristicUuids:characteristic svcId:&svcId charId:&charId];
-        _mBleLayer->HandleWriteConfirmation((__bridge void *) peripheral, &svcId, &charId);
+        dispatch_async(_chipWorkQueue, ^{
+            _mBleLayer->HandleWriteConfirmation((__bridge void *) peripheral, &svcId, &charId);
+        });
     } else {
         ChipLogError(Ble, "BLE:Error writing Characteristics in Chip service on the device: [%@]", error.localizedDescription);
-        _mBleLayer->HandleConnectionError((__bridge void *) peripheral, BLE_ERROR_GATT_WRITE_FAILED);
+        dispatch_async(_chipWorkQueue, ^{
```

Comment:
This one looks safe, by why async, not sync like it used to be?

Reply:
Mostly because I have not seen the need to make it `sync`.  But I may have missed something again.

Reply:
In general you want to async, so you avoid sync lock on queues

Reply:
The main question is a semantic one: does the caller expect that `HandleWriteConfirmation` returns before this function returns?  If it doesn't care, async is fine.